src/System Application/App/Company Initialize/CompanyInitializeImpl.Codeunit.al

```diff
@@ -0,0 +1,51 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Environment.Configuration;
+
+using System.Environment;
+
+codeunit 901 "Company Initialize Impl."
+{
+    Access = Internal;
+
+    procedure InitializeCompany()
+    var
+        CompanyInitialize: Record "Company Initialize";
+        CompanyInitializeCodeunit: Codeunit "Company Initialize";
+        ModuleInfo: ModuleInfo;
+    begin
+        CompanyInitialize."Initialized Time" := CurrentDateTime();
+
+        if NavApp.GetCurrentModuleInfo(ModuleInfo) then
+            CompanyInitialize."Initialized Version" := format(ModuleInfo.AppVersion());
+        CompanyInitialize.Insert();
+
+        CompanyInitializeCodeunit.InitializeCompanySetup();
+
+        CompanyInitializeCodeunit.InitializeCompany();
```

Comment:
What's the gain of raising two identical events (except for their name) in a row?
And shouldn't their names start with `On*`?

Reply:
True, they should probably start with On, my thought is that we have a lot of setup tables that are standalone and doesn't depend on anything else being setup and then we have other tables that does depend on the initial setup tables being filled in. With this you always subscribe to the first event if you're independent of any other setup and second if you are dependent. Of course this does not account for third or fourth level dependencies and there are also many other ways to deal with this kind of problem. It's simply trying to help a very complicated problem a little bit.